operator/pkg/controllers/controlplane/controlplane.go

```diff
@@ -38,10 +37,10 @@ type controlPlane struct {
 }
 
 // NewController returns a controller for managing controlPlane components of the cluster
-func NewController(kubeClient client.Client, account awsprovider.AccountMetadata) *controlPlane {
+func NewController(kubeClient client.Client, account awsprovider.AccountMetadata, iamProvider v1alpha1.ReconcileFinalize) *controlPlane {
 	return &controlPlane{
 		etcdController:   etcd.New(kubeprovider.New(kubeClient)),
-		masterController: master.New(kubeprovider.New(kubeClient), account),
+		masterController: master.New(kubeprovider.New(kubeClient), account, iamProvider),
 		addonsController: addons.New(kubeprovider.New(kubeClient)),
```

Comment:
Also, on a different note, this doesn't make sense to me to be part of controlplane. Any reason why you have added this to controlplane, instead of dataplane ?

Reply:
Addons you mean?